apiserver/apiserver_test.go

```diff
@@ -0,0 +1,332 @@
+package apiserver
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	restful "github.com/emicklei/go-restful"
+	"github.com/pmezard/go-difflib/difflib"
+
+	"istio.io/manager/model"
+	"istio.io/manager/test/mock"
+)
+
+var (
+	routeRuleKey = model.Key{Name: "name", Namespace: "namespace", Kind: "route-rule"}
+
+	validRouteRuleJSON              = []byte(`{"type":"route-rule","name":"name","spec":{"destination":"service.namespace.svc.cluster.local","precedence":1,"route":[{"tags":{"version":"v1"},"weight":25}]}}`)
+	validUpdatedRouteRuleJSON       = []byte(`{"type":"route-rule","name":"name","spec":{"destination":"service.namespace.svc.cluster.local","precedence":1,"route":[{"tags":{"version":"v2"},"weight":25}]}}`)
+	validDiffNamespaceRouteRuleJSON = []byte(`{"type":"route-rule","name":"name","spec":{"destination":"service.differentnamespace.svc.cluster.local","precedence":1,"route":[{"tags":{"version":"v3"},"weight":25}]}}`)
+
+	itemExistsResponse  = "item already exists"
+	invalidTypeResponse = "unknown configuration type not-a-route-rule; use one of [destination-policy ingress-rule route-rule]"
+	notFoundResponse    = "item not found"
```

Comment:
These are expected error strings? I think you should instantiate the error and compare it's `Error()` instead of hardcoding the strings.